crates/rome_js_syntax/src/jsx_ext.rs

```diff
@@ -137,6 +139,51 @@ impl JsxSelfClosingElement {
     }
 }
 
+impl JsxAttributeList {
+    /// Find and return the `JsxAttribute` that matches the given name like [find_attribute_by_name].
+    ///
+    /// Each name of "names_to_lookup" should be unique.
+    /// Support maximum of 16 names.
+    pub fn find_attributes_by_name<const N: usize>(
+        &self,
+        names_to_lookup: [&str; N],
+    ) -> [Option<JsxAttribute>; N] {
+        // assert there are no duplicates
+        debug_assert!(HashSet::<_>::from_iter(names_to_lookup).len() == N);
+        debug_assert!(N <= 16);
+
+        const INIT: Option<JsxAttribute> = None;
+        let mut results = [INIT; N];
+
+        let mut missing = N;
+
+        'attributes: for att in self.iter() {
+            if let Some(attribute) = att.as_jsx_attribute() {
+                if let Some(name) = attribute
+                    .name()
+                    .ok()
+                    .and_then(|x| x.as_jsx_name()?.value_token().ok())
+                {
+                    let name = name.text_trimmed();
+                    for i in 0..N {
+                        if results[i].is_none() && names_to_lookup[i] == name {
+                            results[i] = Some(attribute.clone());
+                            if missing == 1 {
+                                break 'attributes;
+                            } else {
+                                missing -= 1;
+                                break;
+                            }
+                        }
+                    }
```

Comment:
`results[i].is_none()` is guaranteed to be true for as long as the attribute names aren't overlapping (which, in turn, is guaranteed by the debug assertion). The expression can only be `false` if there's a bug in the implementation. That's why I would recommend changing it to a `debug_assert` if it is essential to assert for the condition. 
```suggestion
                    for i in 0..N {
                        if names_to_lookup[i] == name {
                            results[i] = Some(attribute.clone());
                            if missing == 1 {
                                break 'attributes;
                            } else {
                                missing -= 1;
                                break;
                            }
                        }
                    }
```

Reply:
Yes, but checking if the ```Option``` is none is faster than checking the equality of two ```str```.

Reply:
Another difference is:

```js
<a a="1" a="2" />
```

Should we return ```1``` or ```2```? Mine returns ```1```, which is concistent with the other find methods.

Reply:
> Yes, but checking if the Option is none is faster than checking the equality of two str.

In theory. The question is how much is it going to confuse the branch predictor.

Reply:
> Should we return 1 or 2? Mine returns 1, which is concistent with the other find methods.

This is probably worth documenting